ci-operator/step-registry/cucushift/installer/rehearse/aws/cases/clusters/provision/cucushift-installer-rehearse-aws-cases-clusters-provision-commands.sh

```diff
@@ -0,0 +1,741 @@
+  fi
+  return 1
+}
+
+if [ ! -f "${OUT_SELECT}" ]; then
+  echo "ERROR: Not found OUT_SELECT file."
+  exit 1
+fi
+
+if [ ! -f "${OUT_SELECT_DICT}" ]; then
+  echo "ERROR: Not found OUT_SELECT_DICT file."
+  exit 1
+fi
+
+function current_date() { date -u +"%Y-%m-%d %H:%M:%S%z"; }
+
+function update_result() {
+  local k=$1
+  local v=${2:-}
+  cat <<< "$(jq -r --argjson kv "{\"$k\":\"$v\"}" '. + $kv' "$OUT_RESULT")" > "$OUT_RESULT"
+}
+
+function post_actions() {
+  set +e
+
+  current_time=$(date +%s)
+
+  echo "Copying kubeconfig and metadata.json to shared dir"
+  cp \
+      -t "${SHARED_DIR}" \
+      "${INSTALL_DIR}/auth/kubeconfig" \
+      "${INSTALL_DIR}/auth/kubeadmin-password" \
+      "${INSTALL_DIR}/metadata.json"
+
+  echo "Copying log bundle..."
+  cp "${INSTALL_DIR}"/log-bundle-*.tar.gz "${ARTIFACT_DIR}/" 2>/dev/null
+
+  echo "Copying install log and removing REDACTED info from log..."
+  sed '
+    s/password: .*/password: REDACTED/;
+    s/X-Auth-Token.*/X-Auth-Token REDACTED/;
+    s/UserData:.*,/UserData: REDACTED,/;
+    ' "${INSTALL_DIR}/.openshift_install.log" > "${ARTIFACT_DIR}/.openshift_install-${current_time}.log"
+
+  # Writing result
+  # "Bucket": "$(echo "${JOB_SPEC}" | jq -r '.decoration_config.gcs_configuration.bucket')",
+  # "JobUrlPrefix": "$(echo "${JOB_SPEC}" | jq -r '.decoration_config.gcs_configuration.job_url_prefix')",
+  
+  update_result "Region" "${REGION}"
+  update_result "CPType" "${CONTROL_PLANE_INSTANCE_TYPE}"
+  update_result "CPamily" "${CONTROL_PLANE_INSTANCE_TYPE_FAMILY}"
+  update_result "CType" "${COMPUTE_INSTANCE_TYPE}"
+  update_result "CFamily" "${COMPUTE_INSTANCE_TYPE_FAMILY}"
+  update_result "Arch" "${ARCH}"
+  update_result "AMI" "${AMI_RESULT}"
+  update_result "Install" "${INSTALL_RESULT}"
+  update_result "Health" "${HEALTHCHECK_RESULT}"
+  update_result "CreatedDate" "${CREATED_DATE}"
+  update_result "Job" "$(echo "${JOB_SPEC}" | jq -r '.job')"
+  update_result "BuildID" "$(echo "${JOB_SPEC}" | jq -r '.buildid')"
+  update_result "URL" "TBD"
+  update_result "RowUpdated" "$(current_date)"
+
+  echo "RESULT:"
+  jq -r . "${OUT_RESULT}"
+
+  # save JOB_SPEC to ARTIFACT_DIR for debugging
+  echo "${JOB_SPEC}" | jq -r . > ${ARTIFACT_DIR}/JOB_SPEC.json
+
+}
+
+# creating cluster
+
+SSH_PUB_KEY=$(< "${CLUSTER_PROFILE_DIR}/ssh-publickey")
+PULL_SECRET=$(< "${CLUSTER_PROFILE_DIR}/pull-secret")
+
+REGION="$(jq -r '.Region' "${OUT_SELECT_DICT}")"
+ARCH="$(jq -r '.Arch' "${OUT_SELECT_DICT}")"
+
+CONTROL_PLANE_INSTANCE_TYPE="$(jq -r '.CPType' "${OUT_SELECT_DICT}")"
+CONTROL_PLANE_INSTANCE_TYPE_FAMILY="$(jq -r '.CPFamily' "${OUT_SELECT_DICT}")"
+
+COMPUTE_INSTANCE_TYPE="$(jq -r '.CType' "${OUT_SELECT_DICT}")"
+COMPUTE_INSTANCE_TYPE_FAMILY="$(jq -r '.CFamily' "${OUT_SELECT_DICT}")"
+
+if is_empty "$ARCH"; then
+  # Default ARCH is determined by each plarform.
+  # For most of cased, default is arm.
+  # For the resgions which do not support arm64, then set amd64
+  ARCH="arm64"
```

Comment:
This only needs to be set for the first time when getting region from SUMMARY sheet and arch column is empty. Once the region is tested and filled the result back to the sheet, actually ARCH can be obtained directly from sheet for the next running if same row is selected, is that right?

Reply:
Yes, And the row is selected only determined by `CLUSTER_PARAMETER_SELECT_CONDITIONS`.